File: MyAnimeList/src/main/java/com/example/myanimelist/di/modules/DataBaseManagerModule.java
package com.example.myanimelist.di.modules;


import com.example.myanimelist.manager.DataBaseManager;
import dagger.Module;
import dagger.Provides;

import javax.inject.Singleton;

@Module
public class DataBaseManagerModule {
    @Singleton
    @Provides
    DataBaseManager provideDB(){
        return DataBaseManager.getInstance();
    }
}

File: MyAnimeList/src/main/kotlin/com/example/myanimelist/di/modules/DataBaseManagerModule.kt
package com.example.myanimelist.di.modules
import com.example.myanimelist.manager.DataBaseManager
import dagger.*
import javax.inject.Singleton

@Module
class DataBaseManagerModule {

    @Singleton
    @Provides
    fun provideDB() : DataBaseManager{
        return DataBaseManager.getInstance();
    }
}